internal/flow/internal/controller/node_builtin_component.go

```diff
@@ -165,11 +165,13 @@ func NewBuiltinComponentNode(globals ComponentGlobals, reg component.Registratio
 
 func getManagedOptions(globals ComponentGlobals, cn *BuiltinComponentNode) component.Options {
 	cn.registry = prometheus.NewRegistry()
+	parent, id := path.Split(cn.globalID)
 	return component.Options{
 		ID:     cn.globalID,
-		Logger: log.With(globals.Logger, "component", cn.globalID),
+		Logger: log.With(globals.Logger, "component_path", parent, "component_id", id),
 		Registerer: prometheus.WrapRegistererWith(prometheus.Labels{
-			"component_id": cn.globalID,
+			"component_path": parent,
```

Comment:
I think the mixin will need to be updated alongside this, since `component_id` _is_ used in some dashboards (which is why we were doing this prior to 1.0 since it would be a breaking change to split out the labels after-the-fact).

Reply:
I've pushed a commit to include references to `component_path` eg when summing up component-level metrics for the Controller and Prometheus dashboards.

Here's how it looks like in practice. I'll check the dashboards in more environments with nested modules and more data.
![Screenshot 2024-04-02 171206](https://github.com/grafana/agent/assets/17771679/b3233ed8-187f-4442-a88b-ac3577fad49a)


Finally, the OpenTelemetry dashboard doesn't drill down to the component level so it's left as-is for now.